### DISCUSSION TITLE
Themes

### DISCUSSION THREAD

**Discussion by avalon60** [TomSchimansky/CustomTkinter#548] [Showcase]
Hello.

I have developed a few CustomTkinter themes. Hopefully you will find them useful. Feel free to do with them as you will. 

I developed these with  Ctk Theme Builder, an app I have been developing, which I intend to publish on GitHub in the not so distant future. It's at quite an advanced stage and is VERY productive with a WYSIWYG interface. Here are the shots of the themes, as previewed in the app:

GreyGhost:

![image](https://user-images.githubusercontent.com/89534395/197474407-ece00559-e4be-4db7-8f64-e759a3a000b4.png)

NOTE: GreyGhost is the theme used by Ctk Theme Builder.

Oceanix:

![image](https://user-images.githubusercontent.com/89534395/197405608-02ff3222-4b53-4cfa-a9dc-da30563623ae.png)

TrojanBlue:

![image](https://user-images.githubusercontent.com/89534395/197392033-7ab44658-49ab-4681-8fd2-52a3cec0b4fb.png)

MoonlitSky:

![image](https://user-images.githubusercontent.com/89534395/197409606-e1863c3c-1ded-4730-8677-893f8af837d9.png)

Zip File:

[ctk_theme_builder_themes.zip](https://github.com/TomSchimansky/CustomTkinter/files/9847272/ctk_theme_builder_themes.zip)

**Comment by Akascape** [comment]
@avalon60 superb work!!!

**Comment by sumeshir26** [comment]
These themes look nice!
I feel like that there should be a repository where peope can contribute thier themes.

**Reply by avalon60** [reply]
Thanks guys - a lot of love and hard work gone into this. I was inspired by the work done by [@Alkascape](https://github.com/Akascape).

**Comment by TomSchimansky** [comment]
Looks great, must be a lot of work to configure all those values in the right way 😅. I thought of adding a the functionality to load a new theme at runtime, but that's also not that easy.

There will be some additional values in the theme file for new widgets in the next version, and maybe some names will change slightly. But I will list these changes here if the new version is ready. Im looking forward to use this program to create new themes, cause at the moment it's really a pain to edit the theme files manually, its so slow.

**Reply by avalon60** [reply]
Thanks [Tom](https://github.com/TomSchimansky). I need to tidy the code up a little, and there are a couple of impacting issues I've raised with you. As you adjust the colours it instantly updates the associated widget display in the preview panel, hence why I raised those particular issues. I also want to add  widget geometry controls. 

You can create/maintain themes very quickly. It allows you to right click the Palette tiles, and a context menu appears. From there, you can select Copy / Paste / Colour Picker on the Palette colours. The Palette, stores colours on a theme by theme basis, so you can set up some primary and accent colours, independently for each theme. So it's easy to set and copy colours around. You can also right click the tiles in the Control Panel, to get similar functions, or single click any in the lower region of the Palette display to copy a colour. Basically you can copy and paste colours quite freely.

Also I have different views to present widget property groupings. The views are based on JSON files. Anyone who knows JSON and understands the CustomTkinter theme entries, can add their own views, simply by adding a new file in a views folder. 

And yes, as per your comment, I have to reload the Preview Panel (it's a second/server, instance of the same program), when the theme selection is changed. I also reload (for reasons I won't go into), if the appearance mode is toggled. 

When changing / adding property names, it would be good if you could keep the naming structure as consistent as possible. For example, for the most part, I use the leading segment of the property name, to determine the  widget type. However, there is for example, an exception, text_button_disabled. This is a real pain, because I have to explicitly trap that exception to the rule, and It also throws out the display sequence (see Control Panel image for MoonlitSky) 😏

**Reply by Akascape** [reply]
@avalon60 
> I also want to add widget geometry controls.

I am working on a project where you can adjust the geomentry/colors of widgets dynamically.
You can look here: https://github.com/TomSchimansky/CustomTkinter/issues/549

It's almost finished but some widgets have a little issue while resizing.

**Reply by avalon60** [reply]
Yes - that's similar to what I have in mind. I just need to re-jig the real-estate, to fit the options in.

**Comment by abhaymakes** [comment]
Woah @avalon60 those look beautiful. Really an amazing work!

**Comment by OneBigLotus** [comment]
@avalon60 Has there been any updates to the theme builder?

**Reply by avalon60** [reply]
Hi. I  have been busy with work projects and put the theme builder on hold, mainly because it was based on  CustomTkinter version 4. With version 4 the theme JSON, was a radically different format to v5. I was planning to revisit last weekend, and start working on converting to v5 - this work is not trivial! However something cropped up. I will however be resuming work on it within the next week or so.

**Reply by OneBigLotus** [reply]
Oh I'm in my own deep hole of confusion with version 5 right now so I totally understand.
